src/libs/ajax/DataRepo.ts

```diff
@@ -103,6 +104,7 @@ export interface DataRepoContract {
   dataset: (datasetId: string) => {
     details: (include?: DatasetInclude[]) => Promise<DatasetModel>;
     roles: () => Promise<string[]>;
+    createDatasetRequest(_request: DatasetAccessRequestAPI): Promise<DatasetAccessRequestAPI>;
```

Comment:
Although the UI calls this "create dataset request"  the API itself is `createSnapshotRequest` and the API type is `SnapshotBuilderAccessRequest` (in the swagger and generated java in jade-data-repo). Whenever the user terms and implementation terms disagree there's always a problem about where you switch from using UI/user terms and start using API/implementation terms. In this case, I would expect to see the implementation terms used here rather than the user terms.

Reply:
Yeah, my rule is that things defined in `DataRepo.ts` should use implementation terms, things used in `DatasetBuilder` code should be user terms.